### ISSUE TITLE
Add option for generate original type too

### ISSUE THREAD

**Issue by anhdd-kuro** [Code-Hex/graphql-codegen-typescript-validation-schema#107]
I know this plugin produces mutation validators in the first place, but can we have an option for original type as well?
Nowadays we are not only uses zod for mutations, but also validates the data received from the server after fetching.
I know I can just add the ID to the schema using zod's extend, but it would be even more convenient if this plugin could do that.

```ts
type Member {
  id: ID!
  firstName: String
  lastName: String
}

type MemberInput {
  firstName: String
  lastName: String
}

// ->
// add this
export function MemberSchema(): z.ZodObject<Properties<Member>> {
  return z.object({
    id: z.string(),
    firstName: z.string().nullish(),
    lastName: z.string().nullish(),
  })
}

export function MemberInputSchema(): z.ZodObject<Properties<MemberInput>> {
  return z.object({
    firstName: z.string().nullish(),
    lastName: z.string().nullish(),
  })
}
```

so I can
 
```ts
import { graphQLClient, Member, MemberSchema } from "@/infra/graphql";

const getMembers = async (query = defaultQuery): Promise<Member[]> => {
  const { members } = await graphQLClient.request(query);

  const validatedData = MemberSchema().array().parse(members);
  // instead of 
  // const validatedData = MemberInputSchema().extend({ id: z.string() }).array().parse(members);

  return validatedData;
};
```

**Comment by Brian-McBride** [comment]
I needed the same thing. I opened a PR #109

**Comment by Brian-McBride** [comment]
I threw up an NPM package just in case this feature doesn't become part of this one. 

https://www.npmjs.com/package/@anatine/graphql-zod-validation

In your config setting, just add `useObjectTypes: true` and you should get what you are looking for.

**Comment by Code-Hex** [comment]
@anhdd-kuro Thanks for your suggestion!

Previously, I closed the suggestion https://github.com/Code-Hex/graphql-codegen-typescript-validation-schema/issues/20

this was because fragment support was likely to take time. I'd be happy to merge this and any other correspondence if someone could help me with this!

**Comment by Code-Hex** [comment]
I'm watching @Brian-McBride PR #109

**Comment by Code-Hex** [comment]
@Brian-McBride @anhdd-kuro Thanks for great work!!

I've merged PR #132. I will adjust it a little bit and publish it as a new version.

**Comment by Code-Hex** [comment]
@anhdd-kuro @Brian-McBride Hi 👋 

I just published the latest version [v0.6.0](https://github.com/Code-Hex/graphql-codegen-typescript-validation-schema/tree/v0.6.0) which is included this feature.

Thank you!
